lib/saluki-env/src/autodiscovery/providers/local.rs

```diff
@@ -0,0 +1,298 @@
+                        Ok((config_id, config)) => {
+                            found_configs.insert(config_id.clone());
+
+                            // Check if this is a new or updated configuration
+                            if !known_configs.contains(&config_id) {
+                                debug!("New configuration found: {}", config_id);
+
+                                let event = AutodiscoveryEvent { config };
+                                emit_event(sender, event);
+                                known_configs.insert(config_id.clone());
+                            } else {
+                                // Config ID exists, but the content might have changed
+                                debug!("Configuration updated: {}", config_id);
+
+                                let event: AutodiscoveryEvent = AutodiscoveryEvent { config };
+                                emit_event(sender, event);
+                            }
+                        }
+                        Err(e) => {
+                            warn!("Failed to parse config file {}: {}", path.display(), e);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Clean up removed configurations
+    let to_remove: Vec<String> = known_configs
+        .iter()
+        .filter(|config_id| !found_configs.contains(*config_id))
+        .cloned()
+        .collect();
+
+    for config_id in to_remove {
+        debug!("Configuration removed: {}", config_id);
+        known_configs.remove(&config_id);
+
+        // Create an unschedule config
+        let config = Config {
+            name: config_id,
+            event_type: EventType::Unschedule,
+            init_config: Vec::new(),
+            instances: Vec::new(),
+            metric_config: Vec::new(),
+            logs_config: Vec::new(),
+            ad_identifiers: Vec::new(),
+            provider: "local".to_string(),
+            service_id: String::new(),
+            tagger_entity: String::new(),
+            cluster_check: false,
+            node_name: String::new(),
+            source: String::new(),
+            ignore_autodiscovery_tags: false,
+            metrics_excluded: false,
+            logs_excluded: false,
+            advanced_ad_identifiers: Vec::new(),
+        };
+
+        let event = AutodiscoveryEvent { config };
+        emit_event(sender, event);
+    }
+
+    Ok(())
+}
+
+fn emit_event(sender: &Sender<AutodiscoveryEvent>, event: AutodiscoveryEvent) {
+    match sender.send(event) {
+        Ok(_) => (),
+        Err(e) => {
+            warn!("Failed to send autodiscovery event: {}", e);
+        }
+    }
+}
+
+#[async_trait]
+impl AutodiscoveryProvider for LocalAutoDiscoveryProvider {
+    async fn subscribe(&self) -> Receiver<AutodiscoveryEvent> {
+        self.listener_init
+            .get_or_init(|| async {
+                self.start_background_monitor(30).await;
```

Comment:
Similar comment here about lifting up this interval constant to a `const` for discoverability.